drivers/drivers.go

```diff
@@ -267,6 +272,22 @@ func ParseOSURL(input string, useFullAPI bool) (OSDriver, error) {
 		u.Scheme = ""
 		return NewFSDriver(u), nil
 	}
+	if u.Scheme == "w3s" {
+		// W3S URL format: 'w3s://key:proof@pubId/path'
+		// Key and proof are base64url-encoded
+		// pubId must be a unique value used until Publish() is called
+		ucanKey := u.User.Username()
```

Comment:
Updated in deaec76a15db9db3917464376ff151117d2fcdab

Reply:
Looks like the package var for the key is really just a placeholder for now and the actual work to use the key with the w3s tools will be done with https://github.com/livepeer/catalyst/issues/383. I could see a scenario where this package var is not necessary at all and in fact this package does not need to be aware of the private key at all because the private key configuration is all encapsulated in either the script that handles uploads (option 1 in issue 383) or in a single config step (i.e. setup a on-disk config file) for the `w3` CLI tool (option 2 in issue 383). Seems reasonable to nail down the solution for 383 first and then can always cleanup here based on that.

Reply:
> Looks like the package var for the key is really just a placeholder for now and the actual work to use the key with the w3s tools will be done with [livepeer/catalyst#383](https://github.com/livepeer/catalyst/issues/383). I could see a scenario where this package var is not necessary at all and in fact this package does not need to be aware of the private key at all because the private key configuration is all encapsulated in either the script that handles uploads (option 1 in issue 383) or in a single config step (i.e. setup a on-disk config file) for the `w3` CLI tool (option 2 in issue 383). Seems reasonable to nail down the solution for 383 first and then can always cleanup here based on that.

Yes, these are good points. Although, actually, the in-memory variable is kind-of the most secure way of storing the private key, because you don't even store it in any FS. Anyway, let's do it as you suggested, for now leave it as it is and we can revisit it while working on https://github.com/livepeer/catalyst/issues/383